packages/timeline/src/browser/timeline-widget.tsx

```diff
@@ -0,0 +1,208 @@
+import { CommandRegistry } from '@theia/core/lib/common';
+import { TimelineEmptyWidget } from './timeline-empty-widget';
+import { toArray } from '@phosphor/algorithm';
+import URI from '@theia/core/lib/common/uri';
+import { TimelineProvidersChangeEvent } from '../common/timeline-model';
+import { TimelineAggregate } from './timeline-service';
+
+@injectable()
+export class TimelineWidget extends BaseWidget {
+
+    protected panel: Panel;
+    static ID = 'timeline-view';
+
+    private readonly timelinesBySource = new Map<string, TimelineAggregate>();
+
+    @inject(TimelineTreeWidget) protected readonly resourceWidget: TimelineTreeWidget;
+    @inject(TimelineService) protected readonly timelineService: TimelineService;
+    @inject(CommandRegistry) protected readonly commandRegistry: CommandRegistry;
+    @inject(ApplicationShell) protected readonly applicationShell: ApplicationShell;
+    @inject(TimelineEmptyWidget) protected readonly timelineEmptyWidget: TimelineEmptyWidget;
+
+    constructor() {
+        super();
+        this.id = TimelineWidget.ID;
+        this.addClass('theia-timeline');
+    }
+
+    @postConstruct()
+    protected init(): void {
+        const layout = new PanelLayout();
+        this.layout = layout;
+        this.panel = new Panel({
+            layout: new PanelLayout({
+            })
+        });
+        this.panel.node.tabIndex = -1;
+        layout.addWidget(this.panel);
+        this.containerLayout.addWidget(this.resourceWidget);
+        this.containerLayout.addWidget(this.timelineEmptyWidget);
+
+        this.refresh();
+        this.refreshList();
+        this.toDispose.push(this.timelineService.onDidChangeTimeline(event => {
+                const currentWidgetUri = this.getCurrentWidgetUri();
+                if (currentWidgetUri ) {
+                    this.loadTimeline(currentWidgetUri, event.reset);
+                }
+            })
+        );
+        this.toDispose.push(this.applicationShell.onDidChangeCurrentWidget(async event => {
+            const current = event.newValue;
+            if (current && NavigatableWidget.is(current)) {
+                const uri = current.getResourceUri();
+                if (uri) {
+                    this.timelineEmptyWidget.hide();
+                    this.resourceWidget.show();
+                    this.loadTimeline(uri, true);
+                }
+                return;
+            }
+            if (!this.suitableWidgetsOpened()) {
+                this.resourceWidget.hide();
+                this.timelineEmptyWidget.show();
+            }
+        }));
+        this.toDispose.push(this.timelineService.onDidChangeProviders(e => this.onProvidersChanged(e)));
+    }
+
+    private onProvidersChanged(event: TimelineProvidersChangeEvent): void {
+        const current = this.applicationShell.currentWidget;
+        const currentUri = NavigatableWidget.is(current) ? current.getResourceUri() : undefined;
```

Comment:
`const currentUri = this.this.getCurrentWidgetUri();`

Reply:
done